src/instance.rs

```diff
@@ -522,6 +522,19 @@ impl<'a, 'py> Borrowed<'a, 'py, PyAny> {
         Self(NonNull::new_unchecked(ptr), PhantomData, py)
     }
 
+    #[inline]
+    pub(crate) fn downcast<T>(self) -> Result<Borrowed<'a, 'py, T>, DowncastError<'a, 'py>>
```

Comment:
Given that the argument extraction works with `Borrowed` as the primary type, this turned out to be useful to get to `Borrowed<PyString>` for the keyword extraction.

Reply:
Correction: this also enabled `Borrowed<T>` to be used as the holder in `impl PyFunctionArgument for &Bound`.

Reply:
Removed this and used `&Bound` in those places.